src/magma/app/App.cpp

```diff
@@ -148,22 +149,15 @@ void App::createTexture()
         memcpy(data, img.data(), static_cast<size_t>(imageSize));
     vkUnmapMemory(device->handler(), stagingBuffer.mem);
 
-    texture = device->createTexture2D(
-            img.getWidth(), img.getHeight(),
-            VK_FORMAT_R8G8B8A8_SRGB, VK_IMAGE_TILING_OPTIMAL,
-            VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_SAMPLED_BIT,
-            VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT,
-            VK_IMAGE_ASPECT_COLOR_BIT
-    );
-    device->transitionImageLayout(
-            texture.img(), VK_FORMAT_R8G8B8A8_SRGB,
-            VK_IMAGE_LAYOUT_UNDEFINED, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL);
-    device->copyBufferToImage(
-            stagingBuffer.buf, texture.img(),
-            static_cast<uint32_t>(img.getWidth()), static_cast<uint32_t>(img.getHeight()));
-    device->transitionImageLayout(
-            texture.img(), VK_FORMAT_R8G8B8A8_SRGB,
-            VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL);
+    TextureManager& textureManager = device->getTextureManager();
+    texture = textureManager.createTexture2D("input_texture",
+        VK_FORMAT_R8G8B8A8_SRGB,
+        VkExtent3D{(uint)img.getWidth(), (uint)img.getHeight(), 1},
+        VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_SAMPLED_BIT,
+        VK_IMAGE_ASPECT_COLOR_BIT);
+    textureManager.setLayout(texture, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL);
+    textureManager.copyFromBuffer(texture, stagingBuffer.buf);
+    textureManager.setLayout(texture, VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL);
```

Comment:
Actually, all this code should be put into a method like:
```
Texture loadTexture(const std::string &texName, const std::string &path);
```

Reply:
What do you mean?

Reply:
I mean, that process of loading textures from a disk should be encapsulated by a special TextureManager method.